

odometry:
  is_publish_odometry_msgs: true
  scan_matching:
    icp_objective: PointToPlane
    max_correspondence_dist: 0.2
    knn_normal_estimation: 32
    max_n_iter: 30
  scan_processing:
    voxel_size: 0.05
    downsampling_ratio: 1.0
    scan_cropping:
      cropping_radius_max: 15.0
      cropping_radius_min: 2.0
      min_z: -30.0
      max_z: 30.0
      cropper_type: "MinMaxRadius" #options are Cylinder, MaxRadius, MinRadius, MinMaxRadius


mapping:
  is_print_timing_information: true
  is_build_dense_map: false
  is_attempt_loop_closures: false
  is_use_map_initialization: true
  is_merge_scans_into_map: true
  dump_submaps_to_file_before_after_lc: false
  is_refine_odometry_constraints_between_submaps: false
  min_movement_between_mapping_steps: 0.0
  submaps_num_scan_overlap: 1
  scan_to_map_refinement:
    min_refinement_fitness: 0.7
    scan_matching:
      icp_objective: PointToPlane
      max_correspondence_dist: 1.0
      knn_normal_estimation: 1
      max_n_iter: 1
    scan_processing:
      voxel_size: 0.1
      downsampling_ratio: 1.0
      scan_cropping:
        cropping_radius_max: 15.0
        cropping_radius_min: 2.0
        min_z: -20.0
        max_z: 20.0
        cropper_type: "MinMaxRadius" #options are Cylinder, MaxRadius, MinRadius, MinMaxRadius
    
  map_intializer:
    is_initialize_interactively: true
    frame_id: "map_o3d"
    pcd_file_path: "/data/2023-06-22-09-10-12.pcd"
    init_pose:
      position:
        x: 0.0
        y: 0.0
        z: 0.0
      orientation: #in degrees  
        roll: 0.0
        pitch: 0.0
        yaw: 0.0

  submaps:
    size: 30
    min_num_range_data: 10
    adjacency_based_revisiting_min_fitness: 0.4
    
  map_builder:
    scan_cropping:
      cropping_radius_max: 20.0
      cropping_radius_min: 2.0
      min_z: -20.0
      max_z: 20.0
      cropper_type: "MinMaxRadius" #options are Cylinder, MaxRadius, MinRadius, MinMaxRadius
    map_voxel_size: 0.15
    space_carving:
      voxel_size: 0.2
      max_raytracing_length: 10.0
      truncation_distance: 0.4
      carve_space_every_n_scans: 500000000
      min_dot_product_with_normal: 0.3
      
  dense_map_builder:
    scan_cropping:
      cropping_radius_max: 15.0
      cropping_radius_min: 2.0
      min_z: -12.0
      max_z: 12.0
      cropper_type: "MinMaxRadius" #options are Cylinder, MaxRadius, MinRadius, MinMaxRadius
    map_voxel_size: 0.08
    space_carving:
      neigborhood_radius_for_removal: 0.08
      max_raytracing_length: 25.0
      truncation_distance: 0.4
      carve_space_every_n_scans: 5
      min_dot_product_with_normal: 0.0

  place_recognition:
    feature_map_normal_estimation_radius: 1.0
    feature_voxel_size: 0.5
    feature_radius: 2.5
    feature_knn: 100
    feature_normal_knn: 10
    ransac_num_iter: 1000000
    ransac_probability: 0.99
    ransac_model_size: 3
    ransac_max_correspondence_dist: 0.75
    ransac_correspondence_checker_distance: 0.75
    ransac_correspondence_checker_edge_length: 0.5
    ransac_min_corresondence_set_size: 25
    max_icp_correspondence_distance: 0.3
    min_icp_refinement_fitness: 0.5
    dump_aligned_place_recognitions_to_file: false   
    consistency_check:
      max_drift_roll: 70.0 #deg
      max_drift_pitch: 70.0 # deg
      max_drift_yaw: 70.0 #deg


  global_optimization:
    edge_prune_threshold: 0.2
    loop_closure_preference: 2.0
    max_correspondence_distance: 10.0
    reference_node: 0

visualization:
  assembled_map_voxel_size: 0.1
  submaps_voxel_size: 0.1
  visualize_every_n_msec: 300.0
  
  
saving_parameters:
  save_at_mission_end: true
  save_map: true
  save_submaps: false


  
    
